module/os/linux/spl/spl-taskq.c

```diff
@@ -1041,23 +1050,38 @@ taskq_create(const char *name, int nthreads, pri_t pri,
 	if (flags & TASKQ_THREADS_CPU_PCT) {
 		ASSERT(nthreads <= 100);
 		ASSERT(nthreads >= 0);
-		nthreads = MIN(nthreads, 100);
-		nthreads = MAX(nthreads, 0);
-		nthreads = MAX((num_online_cpus() * nthreads) / 100, 1);
+		proper_nthreads = MIN(nthreads, 100);
+		proper_nthreads = MAX(proper_nthreads, 0);
+		proper_nthreads = MAX((num_online_cpus() * proper_nthreads) /
+		    100, 1);
 	}
 
 	tq = kmem_alloc(sizeof (*tq), KM_PUSHPAGE);
 	if (tq == NULL)
 		return (NULL);
 
+#ifdef HAVE_CPU_HOTPLUG
+	if (nthreads == boot_ncpus || flags & TASKQ_THREADS_CPU_PCT) {
+		tq->tq_hp_support = B_TRUE;
+		if (cpuhp_state_add_instance_nocalls(spl_taskq_cpuhp_state,
+		    &tq->tq_hp_cb_node) != 0) {
+			kmem_free(tq, sizeof (*tq));
+			return (NULL);
+		}
+	} else {
+		tq->tq_hp_support = B_FALSE;
```

Comment:
I think we want to minimize the amount of code / data structures that are different on different kernels (i.e. whether we HAVE_CPU_HOTPLUG or not).  So I agree with having tq_hp_support always exist.

Therefore I think we should initialize it to FALSE even if we don't HAVE_CPU_HOTPLUG.  e.g. 
```
tq_hp_support = B_FALSE;
#ifdef HAVE_CPU_HOTPLUG
if (...CPU_PCT) {
   tq_hp_support = B_TRUE;
```